lib/src/pick.dart

```diff
@@ -40,38 +46,88 @@ class _FolderPickerState extends State<FolderPicker> {
 
   @override
   Widget build(BuildContext context) {
+    bodyList = ListView(children: _getDirs(context));
     return Scaffold(
-      body: ListView(
-        children: (_currentDirectory == rootDirectory)
-            ? _getDirs(context)
-            : [
-                GestureDetector(
-                  child: ListTile(
-                    leading: const Icon(Icons.arrow_upward),
-                    title: const Text("..", textScaleFactor: 1.5),
-                  ),
-                  onTap: () => setState(
-                      () => _currentDirectory = _currentDirectory.parent),
-                )
-              ]
-          ..addAll(_getDirs(context)),
```

Comment:
Every time the user goes up on the folder tree, we check if the upper folder is accesible, if it's not, now the current folder is the root folder and the user won't be able to go up, thus preventing the "Acces Denied" red screen.